--- sblim-cmpi-smbios-0.3.2/cmpiLinux_BIOSProduct.c.orig	2004-12-07 10:45:10.000000000 +0100
+++ sblim-cmpi-smbios-0.3.2/cmpiLinux_BIOSProduct.c	2015-01-17 09:01:48.272039058 +0100
@@ -53,15 +53,15 @@
 
 
 CMPIStatus Linux_BIOSProductCleanup( CMPIInstanceMI * mi, 
-  CMPIContext * ctx) {
+  const CMPIContext * ctx, CMPIBoolean terminating) {
   cimdmi_term();
   CMReturn(CMPI_RC_OK);
 }
 
 CMPIStatus Linux_BIOSProductEnumInstanceNames( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * ref) { 
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * ref) { 
   CMPIStatus rc = {CMPI_RC_OK, NULL};
   CMPIObjectPath *op;
   DMI_BIOSPRODUCT *biosprod;
@@ -89,10 +89,10 @@
 }
 
 CMPIStatus Linux_BIOSProductEnumInstances( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * ref, 
-  char ** properties) { 
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * ref, 
+  const char ** properties) { 
   CMPIStatus rc = {CMPI_RC_OK, NULL};
   CMPIInstance *inst;
   DMI_BIOSPRODUCT *biosprod;
@@ -120,10 +120,10 @@
 }
 
 CMPIStatus Linux_BIOSProductGetInstance( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * cop, 
-  char ** properties) {
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * cop, 
+  const char ** properties) {
   CMPIStatus rc = {CMPI_RC_OK, NULL};
   CMPIInstance *inst;
   DMI_BIOSPRODUCT *biosprod;
@@ -151,10 +151,10 @@
 }
 
 CMPIStatus Linux_BIOSProductCreateInstance( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * cop, 
-  CMPIInstance * ci) {
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * cop, 
+  const CMPIInstance * ci) {
   CMPIStatus rc = {CMPI_RC_OK, NULL};
   if( _debug )
     fprintf( stderr, "--- %s.c : CMPI CreateInstance()\n", _ClassName ); 
@@ -164,11 +164,11 @@
 }
 
 CMPIStatus Linux_BIOSProductSetInstance( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * cop,
-  CMPIInstance * ci, 
-  char **properties) {
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * cop,
+  const CMPIInstance * ci, 
+  const char **properties) {
   CMPIStatus rc = {CMPI_RC_OK, NULL};
   if( _debug )
     fprintf( stderr, "--- %s.c : CMPI SetInstance()\n", _ClassName ); 
@@ -178,9 +178,9 @@
 }
 
 CMPIStatus Linux_BIOSProductDeleteInstance( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * cop) {
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * cop) {
   CMPIStatus rc = {CMPI_RC_OK, NULL}; 
   if( _debug )
     fprintf( stderr, "--- %s.c : CMPI DeleteInstance()\n", _ClassName ); 
@@ -190,11 +190,11 @@
 }
 
 CMPIStatus Linux_BIOSProductExecQuery( CMPIInstanceMI * mi, 
-  CMPIContext * ctx, 
-  CMPIResult * rslt, 
-  CMPIObjectPath * ref, 
-  char * lang, 
-  char * query) {
+  const CMPIContext * ctx, 
+  const CMPIResult * rslt, 
+  const CMPIObjectPath * ref, 
+  const char * query, 
+  const char * lang) {
   CMPIStatus rc = {CMPI_RC_OK, NULL};
   if( _debug )
     fprintf( stderr, "--- %s.c : CMPI ExecQuery()\n", _ClassName ); 
